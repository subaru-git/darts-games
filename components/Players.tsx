import React, { FC, useState } from "react";
import {
  Editable,
  EditableInput,
  EditablePreview,
  Table,
  TableContainer,
  Th,
  Thead,
  Tr,
} from "@chakra-ui/react";

type PlayersProps = {
  players: string[];
  onUpdateName: (names: string[]) => void;
};

const Players: FC<PlayersProps> = ({ players, onUpdateName }) => {
  return (
    <>
      <TableContainer>
        <Table variant="simple">
          <Thead>
            <Tr>
              <Th>
                <Editable
                  defaultValue={players[0]}
                  onChange={(e) => {
                    onUpdateName([e, players[1]]);
                  }}
                >
                  <EditablePreview />
                  <EditableInput />
                </Editable>
              </Th>
              <Th>
                <Editable
                  defaultValue={players[1]}
                  onChange={(e) => {
                    onUpdateName([players[0], e]);
                  }}
                >
                  <EditablePreview />
                  <EditableInput />
                </Editable>
              </Th>
            </Tr>
          </Thead>
        </Table>
      </TableContainer>
    </>
  );
};

export default Players;
