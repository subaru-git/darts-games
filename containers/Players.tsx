import React, { FC, useState } from "react";
import {
  Editable,
  EditableInput,
  EditablePreview,
  Table,
  TableContainer,
  Th,
  Thead,
  Tr,
} from "@chakra-ui/react";

type PlayersProps = {
  players: string[];
  onUpdateName: (names: string[]) => void;
};

const Players: FC<PlayersProps> = ({ players, onUpdateName }) => {
  const [names, setNames] = useState(players);
  return (
    <>
      <TableContainer>
        <Table variant="simple">
          <Thead>
            <Tr>
              <Th>
                <Editable
                  defaultValue={names[0]}
                  onChange={(e) => {
                    setNames([e, names[1]]);
                    onUpdateName([e, names[1]]);
                  }}
                >
                  <EditablePreview />
                  <EditableInput />
                </Editable>
              </Th>
              <Th>
                <Editable
                  defaultValue={names[1]}
                  onChange={(e) => {
                    setNames([names[0], e]);
                    onUpdateName([names[0], e]);
                  }}
                >
                  <EditablePreview />
                  <EditableInput />
                </Editable>
              </Th>
            </Tr>
          </Thead>
        </Table>
      </TableContainer>
    </>
  );
};

export default Players;
