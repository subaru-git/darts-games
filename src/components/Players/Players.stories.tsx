/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import Players from './Players'

export default {
  title: 'Components/Players',
  component: Players,
} as ComponentMeta<typeof Players>

const Template: ComponentStory<typeof Players> = (args) => <Players {...args} />

export const Default = Template.bind({})
Default.args = {
  players: ['Player 1', 'Player 2'],
}
