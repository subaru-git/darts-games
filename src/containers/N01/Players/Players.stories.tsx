import type { Meta, StoryObj } from '@storybook/react';
import Players from './Players';

const meta: Meta<typeof Players> = { component: Players, title: 'Containers/N01/Players' };

export default meta;
type Story = StoryObj<typeof Players>;
export const Default: Story = { args: { players: ['Player 1', 'Player 2'] } };
