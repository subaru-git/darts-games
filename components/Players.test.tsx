import Players from "./Players";
import { fireEvent, render, screen } from "@testing-library/react";

test("should rendering", () => {
  const updateName = jest.fn();
  const { container } = render(
    <Players players={["Player1", "Player2"]} onUpdateName={updateName} />
  );
  expect(screen.getAllByText("Player1")).toHaveLength(1);
  expect(screen.getAllByText("Player2")).toHaveLength(1);
  fireEvent.change(screen.getByDisplayValue("Player1"), {
    target: { value: "Player3" },
  });
  expect(updateName).toBeCalledTimes(1);
  expect(container).toMatchSnapshot();
});
