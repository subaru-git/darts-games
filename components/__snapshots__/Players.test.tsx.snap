// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="chakra-table__container css-0"
  >
    <table
      class="chakra-table css-0"
      role="table"
    >
      <thead
        class="css-0"
      >
        <tr
          class="css-0"
          role="row"
        >
          <th
            class="css-0"
          >
            <div
              class="chakra-editable css-0"
            >
              <span
                class="chakra-editable__preview css-0"
                tabindex="0"
              >
                Player3
              </span>
              <input
                class="chakra-editable__input css-0"
                hidden=""
                value="Player3"
              />
            </div>
          </th>
          <th
            class="css-0"
          >
            <div
              class="chakra-editable css-0"
            >
              <span
                class="chakra-editable__preview css-0"
                tabindex="0"
              >
                Player2
              </span>
              <input
                class="chakra-editable__input css-0"
                hidden=""
                value="Player2"
              />
            </div>
          </th>
        </tr>
      </thead>
    </table>
  </div>
</div>
`;
